webapp/components/Editor/defaultExtensions.spec.js

```diff
@@ -0,0 +1,93 @@
+import defaultExtensions from './defaultExtensions.js'
+import { Editor } from 'tiptap'
+
+let content
+let createEditor
+
+describe('defaultExtensions', () => {
+  describe('editor', () => {
+    createEditor = () => {
+      const componentStub = {
+        placeholder: 'placeholder',
+        $t: jest.fn(),
+        $apollo: {
+          mutate: jest.fn(),
+        },
+      }
+      return new Editor({
+        content,
+        extensions: [...defaultExtensions(componentStub)],
+      })
+    }
+  })
+
+  it('renders', () => {
+    content = ''
+    expect(createEditor().getHTML()).toEqual('<p></p>')
+  })
+
+  describe('`content` contains a mentioning', () => {
+    beforeEach(() => {
+      content =
+        '<p>This is a post content mentioning <a class="mention" data-mention-id="alicias-id" href="/profile/f0628376-e692-4167-bdb4-d521de5a014f" target="_blank">@alicia-luettgen</a>.</p>'
+    })
+
+    it('renders mentioning as link', () => {
+      const editor = createEditor()
+      const expected =
+        '<p>This is a post content mentioning <a href="/profile/f0628376-e692-4167-bdb4-d521de5a014f" rel="noopener noreferrer nofollow">@alicia-luettgen</a>.</p>'
+      expect(editor.getHTML()).toEqual(expected)
+    })
+  })
+
+  describe('`content` contains a hashtag', () => {
+    beforeEach(() => {
+      content =
+        '<p>This is a post content with a hashtag <a class="hashtag" href="/search/hashtag/metoo" target="_blank">#metoo</a>.</p>'
+    })
+
+    it('renders hashtag as link', () => {
+      const editor = createEditor()
+      const expected =
+        '<p>This is a post content with a hashtag <a href="/search/hashtag/metoo" rel="noopener noreferrer nofollow">#metoo</a>.</p>'
+      expect(editor.getHTML()).toEqual(expected)
+    })
+  })
+
+  describe('`content` contains embed code', () => {
+    beforeEach(() => {
+      content =
+        '<p>Baby loves cat: </p><a href="https://www.youtube.com/watch?v=qkdXAtO40Fo" class="embed" target="_blank"></a>'
+    })
+
+    it('recognizes embed code', () => {
+      const editor = createEditor()
+      const expected = {
+        content: [
+          {
+            content: [
+              {
+                text: 'Baby loves cat:',
+                type: 'text',
+              },
+            ],
+            type: 'paragraph',
+          },
+          {
+            content: [
+              {
+                attrs: {
+                  dataEmbedUrl: 'https://www.youtube.com/watch?v=qkdXAtO40Fo',
+                },
+                type: 'embed',
+              },
+            ],
+            type: 'paragraph',
+          },
+        ],
+        type: 'doc',
+      }
+      expect(editor.getJSON()).toEqual(expected)
+    })
+  })
+})
```

Comment:
> [<img alt="mattwr18" height="40" width="40" align="left" src="https://avatars1.githubusercontent.com/u/38261864?s=88&v=4">](https://github.com/mattwr18) **Authored by [mattwr18](https://github.com/mattwr18)**
_<time datetime="2019-08-01T06:28:12Z" title="Thursday, August 1st 2019, 8:28:12 am +02:00">Aug 1, 2019</time>_

---
> **Outdated (history rewrite)** - original diff
---
```diff
@@ -0,0 +1,88 @@
+import defaultExtensions from './defaultExtensions.js'
+import { Editor } from 'tiptap'
+
+let content
+let createEditor
+
+describe('defaultExtensions', () => {
+  describe('editor', () => {
+    createEditor = () => {
+      const componentStub = {
+        placeholder: 'placeholder',
+        $t: jest.fn(),
+        $apollo: {
+          mutate: jest.fn(),
+        },
+      }
+      return new Editor({
+        content,
+        extensions: [...defaultExtensions(componentStub)],
+      })
+    }
+  })
+
+  it('renders', () => {
+    content = ''
+    expect(createEditor().getHTML()).toEqual('<p></p>')
+  })
+
+  describe('`content` contains a mentioning', () => {
+    beforeEach(() => {
+      content =
+        '<p>This is a post content mentioning <a class="mention" href="/profile/f0628376-e692-4167-bdb4-d521de5a014f" target="_blank">@alicia-luettgen</a>.</p>'
+    })
+
+    it('renders mentioning as link', () => {
+      const editor = createEditor()
+      const expected =
+        '<p>This is a post content mentioning <a href="/profile/f0628376-e692-4167-bdb4-d521de5a014f" rel="noopener noreferrer nofollow">@alicia-luettgen</a>.</p>'
+      expect(editor.getHTML()).toEqual(expected)
+    })
+  })
+
+  describe('`content` contains a hashtag', () => {
+    beforeEach(() => {
+      content =
+        '<p>This is a post content with a hashtag <a class="hashtag" href="/search/hashtag/metoo" target="_blank">#metoo</a>.</p>'
+    })
+
+    it('renders hashtag as link', () => {
+      const editor = createEditor()
+      const expected =
+        '<p>This is a post content with a hashtag <a href="/search/hashtag/metoo" rel="noopener noreferrer nofollow">#metoo</a>.</p>'
+      expect(editor.getHTML()).toEqual(expected)
+    })
+  })
+
+  describe('`content` contains embed code', () => {
+    beforeEach(() => {
+      content =
+        '<p>Baby loves cat: </p><a href="https://www.youtube.com/watch?v=qkdXAtO40Fo" class="embed" target="_blank"></a>'
+    })
+
+    it('recognizes embed code', () => {
+      const editor = createEditor()
+      const expected = {
+        content: [
+          {
+            content: [
+              {
+                text: 'Baby loves cat:',
+                type: 'text',
+              },
+            ],
+            type: 'paragraph',
+          },
+          {
+            attrs: {
+              dataEmbedUrl: 'https://www.youtube.com/watch?v=qkdXAtO40Fo',
+            },
+            type: 'embed',
+          },
+        ],
+        type: 'doc',
+      }
+      expect(editor.getJSON()).toEqual(expected)
+    })
+  })
+})
```


really great that you added more tests for the `Editor`!!

Reply:
> [<img alt="roschaefer" height="40" width="40" align="left" src="https://avatars1.githubusercontent.com/u/38261864?s=88&v=4">](https://github.com/roschaefer) **Authored by [roschaefer](https://github.com/roschaefer)**
_<time datetime="2019-08-01T10:18:00Z" title="Thursday, August 1st 2019, 12:18:00 pm +02:00">Aug 1, 2019</time>_

---


Not testing the `Editor.vue` component. Only `tiptap`'s `editor`. Nevertheless, I hope this tests gives us some assurance about how HTML get parsed for `Editor.vue` and `ContentViewer.vue`.